python/tests/e2e/conftest.py

```diff
@@ -132,9 +152,81 @@ def sanitize_request(request: VCRRequest) -> VCRRequest:
         if header in request.headers:
             request.headers[header] = ["<filtered>"]
 
+    if "oauth2.googleapis.com/token" in request.uri and request.body:
+        body_str = (
+            request.body.decode() if isinstance(request.body, bytes) else request.body
+        )
+        body_str = re.sub(r"refresh_token=[^&]+", "refresh_token=<filtered>", body_str)
+        body_str = re.sub(r"client_secret=[^&]+", "client_secret=<filtered>", body_str)
+        request.body = body_str
+
     return request
 
 
+def sanitize_response(response: dict[str, Any]) -> dict[str, Any]:
+    """Sanitize sensitive tokens in VCR response before recording to cassette.
+
+    This hook is called AFTER the real HTTP response is received,
+    but BEFORE it's written to the cassette file. We sanitize OAuth tokens
+    from Google OAuth2 responses (used by AnthropicVertex).
+
+    Args:
+        response: VCR response dict to sanitize
+
+    Returns:
+        Sanitized copy of the response safe for cassette storage
+    """
+    response = deepcopy(response)
+
+    if "body" in response and "string" in response["body"]:
+        body_str = response["body"]["string"]
+        if isinstance(body_str, bytes):
+            try:
+                body_str = body_str.decode()
+            except UnicodeDecodeError:
+                # Body is likely compressed (gzip) or binary data
+                # Skip sanitization for these responses
+                return response
```

Comment:
hypothetically if we get gzipped data with secrets wouldn't we still want to decompress and sanitize it? (and then re-compress before writing it back).